### PR TITLE
Implement Milnor fiber and Coxeter complexes

### PR DESCRIPTION
**Pull request by tscrim**
Implements Milnor fiber and Coxeter complexes and their associated posets. #34936

Dependencies: #34912

### CODE REVIEW

**Comment by tscrim** [comment]
@fchapoton Would you be willing to review this?

**Comment by fchapoton** [comment]
I can try, but I cannot really check the code.

**Comment by tscrim** [comment]
Thank you. No problem if you can't. I can ask Vic Reiner to see if he would be willing to review the mathematical portion. I can also split off the Coxeter group portion too.

File: src/sage/categories/finite_coxeter_groups.py [hunk 1]
```diff
@@ -754,6 +754,152 @@ def permutahedron(self, point=None, base_ring=None):
...
+                    gp = g
+                    D = gp.descents(side='right')
+                    if D and D[0] == i:
+                        D.pop(0)
+                    while D:
+                        gp = gp.apply_simple_reflection(D[0])
+                        D = gp.descents(side='right')
+                        if D and D[0] == i:
+                            D.pop(0)
+                    Ip = list(I)
```
Comment (fchapoton) [inline src/sage/categories/finite_coxeter_groups.py]: maybe store the tuples Ip[i] in a list before the main loop over self ?

Comment (tscrim) [reply src/sage/categories/finite_coxeter_groups.py]: Good idea. Done.

**Comment by fchapoton** [comment]
one linter is not green, about E305

**Comment by tscrim** [comment]
> one linter is not green, about E305

Done, although the linter spec is not well-defined here as it is about blanklines before a class.

**Comment by fchapoton** [comment]
Looks good. Is there no way to avoid looping over all elements ?

Too bad that the posets gets huge very fast, so that one cannot even hope to reach H4 (62881 elts) or D5 (12483 elts). They seem to have nice Coxeter polynomials, by the way.

**Comment by tscrim** [comment]
The number of maximal elements is equal to the number of elements in the group, so we have to iterate over all elements in some fashion. I might be able to rework things so that we are building the group elements out, but this seems more technical and I am not convinced it will be faster in the end.

**Review by fchapoton — APPROVED** [review]
ok, let's get this in.

**Comment by tscrim** [comment]
Thank you!

**Pull request merged**